pkg/reconciler/taskrun/taskrun.go

```diff
@@ -203,28 +204,40 @@ func (c *Reconciler) updateStatusLabelsAndAnnotations(tr, original *v1alpha1.Tas
 	return nil
 }
 
-func (c *Reconciler) getTaskFunc(tr *v1alpha1.TaskRun) (resources.GetTask, v1alpha1.TaskKind) {
-	var gtFunc resources.GetTask
-	kind := v1alpha1.NamespacedTaskKind
+func (c *Reconciler) getTaskFunc(tr *v1alpha1.TaskRun) (resources.GetTask, v1alpha1.TaskKind, error) {
+	if tr.Spec.TaskRef != nil && tr.Spec.TaskRef.Image != "" {
+		taskFunc, err := remote.ImageTaskResolver(c.KubeClientSet, tr.Spec.TaskRef.Image, tr.GetNamespace(), tr.Spec.ServiceAccountName)
+		// This is the only real error case. If we know we need a GetTask func that works on Images but we couldn't create
+		// one, the Task lookup will never suceed
+		if err != nil {
+			return nil, "", fmt.Errorf("Could not lookup remote task: %w", err)
+		}
+
+		kind := v1alpha1.NamespacedTaskKind
+		if tr.Spec.TaskRef.Kind == v1alpha1.ClusterTaskKind {
+			kind = v1alpha1.ClusterTaskKind
+		}
```

Comment:
total nit but i wonder if this logic could be moved out of this if statement and work for all 3 cases somehow